### PR TITLE
homework 3

### PR DESCRIPTION
**Pull request by lesnata**
please check

### CODE REVIEW

File: python_functionality/homework.py [hunk 1]
```diff
@@ -14,7 +14,8 @@ def task_1_fix_names_start_letter(data: DT) -> DT:
         fix_names_start_letters([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}])
         >>> [{'name': 'Alex', 'age': 26}, {'name': 'Denys', 'age': 89}]
     """
-    pass
+    return [{i: (name.capitalize() if isinstance(name, str) else name) for (i, name) in dictionary.items()}
+            for dictionary in data]
```
Comment (alchemixXx) [inline python_functionality/homework.py]: you check if name is string. it's better to check if name exist in object.keys. Look at .get(key, default) method

File: python_functionality/homework.py [hunk 2]
```diff
@@ -25,7 +26,7 @@ def task_2_remove_dict_fields(data: DT, redundant_keys: List[str]) -> DT:
        remove_dict_field([{'name': 'Alex', 'age': 26}, {'name': 'denys', 'age': 89}], 'age')
         >>> [{'name': 'Alex'}, {'name': 'denys'}]
     """
-    pass
+    return [{i: j for (i, j) in dictionary.items() if i not in redundant_keys} for dictionary in data]
```
Comment (alchemixXx) [inline python_functionality/homework.py]: great solution!

File: python_functionality/homework.py [hunk 3]
```diff
@@ -35,21 +36,30 @@ def task_3_find_item_via_value(data: DT, value) -> DT:
...
         >>> [{'name': 'Alex', 'age': 26}]
     """
-    pass
+    new_dict = {}
+    for i, j in data.items():
+        if i == value:
+            new_pair[i] = j
+            new_dict.append(new_pair)
+
+    return new_dict
```
Comment (alchemixXx) [inline python_functionality/homework.py]: you can do it simpler and with list comprehension!

File: python_functionality/homework.py [hunk 4]
```diff
@@ -58,21 +68,25 @@ def task_6_min_value_list_of_dicts(data: DT, key: str) -> ST:
     Returns:
 
     """
-    pass
+    return min([dct for dct in data if key in dct.keys()], key=(lambda d: d[key]))
```
Comment (alchemixXx) [inline python_functionality/homework.py]: good job!

File: python_functionality/homework.py [hunk 5]
```diff
@@ -58,21 +68,25 @@ def task_6_min_value_list_of_dicts(data: DT, key: str) -> ST:
...
 def task_7_max_value_list_of_lists(data: List[List[int]]) -> int:
     """
     Find max value from list of lists
     """
-    pass
+    for i in data:
+        i = max(List)
+        clean_list.append(i)
+
+    return max(clean_list)
```
Comment (alchemixXx) [inline python_functionality/homework.py]: do it with LC and without global object List

File: python_functionality/homework.py [hunk 6]
```diff
@@ -102,4 +116,6 @@ def task_10_generator_of_simple_numbers() -> Generator[int, None, None]:
         next(a)
         >>> 3
     """
-    pass
+    for number in range(2, 201):
+        if all(number % i != 0 for i in range(2, number)):
+            yield number
```
Comment (alchemixXx) [inline python_functionality/homework.py]: nice!

File: python_functionality/homework.py (hunk 1 above)
Comment (lesnata) [reply python_functionality/homework.py]: UPDATED: 
 return [{i: (name.capitalize() if dict.get(name) else name) for (i, name) in dictionary.items()}
            for dictionary in data]

File: python_functionality/homework.py (hunk 3 above)
Comment (lesnata) [reply python_functionality/homework.py]: Would it be better?
return {i:j for (i,j) in data.items() if i == value}

File: python_functionality/homework.py (hunk 5 above)
Comment (lesnata) [reply python_functionality/homework.py]: Like this?
return max([clean_list for default_list in data for clean_list in default_list])

**Pull request merged**